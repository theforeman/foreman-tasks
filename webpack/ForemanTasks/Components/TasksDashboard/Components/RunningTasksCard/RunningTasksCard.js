import React from 'react';
import { translate as __ } from 'foremanReact/common/I18n';

import TasksDonutCard from '../TasksDonutCard/TasksDonutCard';

const RunningTasksCard = ({ ...props }) => (
  <TasksDonutCard title={__('Running')} {...props} />
);

const filterTitle = obj => {
  const { title, ...newObj } = obj;
  return newObj;
};

RunningTasksCard.propTypes = filterTitle(TasksDonutCard.propTypes);
RunningTasksCard.defaultProps = filterTitle(TasksDonutCard.defaultProps);

export default RunningTasksCard;
