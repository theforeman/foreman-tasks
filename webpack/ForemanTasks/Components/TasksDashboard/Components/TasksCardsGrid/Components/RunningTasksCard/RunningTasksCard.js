import React from 'react';
import { translate as __ } from 'foremanReact/common/I18n';

import TasksDonutCard from '../TasksDonutCard/TasksDonutCard';

const RunningTasksCard = ({ ...props }) => (
  <TasksDonutCard
    title={__('Running')}
    wantedQueryState="running"
    id="running-tasks-card"
    {...props}
  />
);

const filterUnwantedFields = obj => {
  const { title, wantedQueryState, ...newObj } = obj;
  return newObj;
};

RunningTasksCard.propTypes = filterUnwantedFields(TasksDonutCard.propTypes);
RunningTasksCard.defaultProps = filterUnwantedFields(
  TasksDonutCard.defaultProps
);

export default RunningTasksCard;
