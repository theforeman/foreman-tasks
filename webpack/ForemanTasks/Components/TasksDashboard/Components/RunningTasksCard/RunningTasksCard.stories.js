import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, number, text, select } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { withCardsDecorator } from '../../../../../stories/decorators';

import {
  TASKS_DASHBOARD_AVAILABLE_QUERY_MODES,
  TASKS_DASHBOARD_AVAILABLE_TIMES,
} from '../../TasksDashboardConstants';
import RunningTasksCard from './RunningTasksCard';

storiesOf('TasksDashboard', module)
  .addDecorator(withKnobs)
  .addDecorator(withCardsDecorator)
  .add('RunningTasksCard', () => {
    const selectTime = select(
      'time',
      TASKS_DASHBOARD_AVAILABLE_TIMES,
      RunningTasksCard.defaultProps.time
    );

    const selectMode = select(
      'mode',
      { ...TASKS_DASHBOARD_AVAILABLE_QUERY_MODES, NONE: 'none' },
      'none'
    );
    return (
      <RunningTasksCard
        data={{
          last: number('data.last', 3),
          older: number('data.older', 5),
        }}
        time={selectTime}
        query={{
          state: text('query.state', 'some-state'),
          mode: selectMode,
          time: selectTime,
        }}
        updateQuery={action('updateQuery')}
      />
    );
  });
