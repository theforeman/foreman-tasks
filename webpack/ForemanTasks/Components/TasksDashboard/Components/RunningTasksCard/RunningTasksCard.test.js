import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import RunningTasksCard from './RunningTasksCard';

const fixtures = {
  'render with minimal props': {},
  'render with some props': { some: 'prop' },
};

describe('RunningTasksCard', () =>
  testComponentSnapshotsWithFixtures(RunningTasksCard, fixtures));
