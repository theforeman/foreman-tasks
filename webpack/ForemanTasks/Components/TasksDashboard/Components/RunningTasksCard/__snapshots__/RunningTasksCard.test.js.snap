// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RunningTasksCard render with minimal props 1`] = `
<TasksDonutCard
  className=""
  data={
    Object {
      "last": 0,
      "older": 0,
    }
  }
  query={Object {}}
  time="H24"
  title="Running"
  updateQuery={[Function]}
  wantedQueryState="running"
/>
`;

exports[`RunningTasksCard render with some props 1`] = `
<TasksDonutCard
  className=""
  data={
    Object {
      "last": 0,
      "older": 0,
    }
  }
  query={Object {}}
  some="prop"
  time="H24"
  title="Running"
  updateQuery={[Function]}
  wantedQueryState="running"
/>
`;
